configurations/dockerDataTemplate.go

```diff
@@ -117,3 +121,27 @@ func (cmdObj cmd) WriteInstruction() string {
 	result := fmt.Sprintf("CMD %s", execFormString)
 	return result
 }
+
+// Programming Language installation instruction.
+type programmingLanguage struct {
+	Name    string
+	Version string
+}
+
+// Gives RUN instruction for installing the specified programmingLanguage.
+func (langObj programmingLanguage) WriteInstruction() string {
+
+	scriptName := fmt.Sprintf("%s_%s.sh", langObj.Name, langObj.Version)
+
+	// Check whether the given language and version are available in the installation scripts.
+	currentDir, _ := os.Getwd()
+	scriptPath := filepath.Join(currentDir, constants.InstallationScriptsDir, scriptName)
+	_, err := os.Stat(scriptPath)
+	if err == nil {
+		result := fmt.Sprintf("RUN ./%s/%s ", constants.InstallationScriptsDir, scriptName)
+		return result
+	} else if os.IsNotExist(err) {
+		log.Fatalf("Installation scripts for given language and version doesn't exists")
```

Comment:
No log fatals as it can result in an unsafe state.
Always perform graceful shutdown and propagate errors back to the source.

Reply:
Yes. I have tried to follow this elsewhere, but the `WriteInstruction()` function is a part of `instruction` which is used as a template for generating docker file template. Hence all the structs that implement `WriteInstruction` return a string only.  Hence I kept the log statement.